src/replication.c

```diff
@@ -1259,15 +1506,65 @@ update_replica_entry(spec_t *spec, replica_t *replica, int iofd)
 		    replica->zvol_guid);
 		goto error_out_replica;
 	}
+
+	known_replica = is_known_replica(spec, replica);
+
+	//TODO: There is a case where user scaled up the replcias (created new cvr)
+	// but he did not update the desired replication factor in that time one of the
+	// known replica is not connected.(Handeled but need to cross verify)
+
+	/* Below snippet is to update known replica list if replica is not present
+	 * known replica list by making call to volume mgmt on succcess update inmemory
+	 * data structure.
+	 */
+	if (known_replica && !is_replica_exist(spec, replica->replica_id)) {
+		// update known replcia list
+		rc = update_known_replica_list(spec, replica);
+		if (rc < 0) {
+			REPLICA_ERRLOG("Failed to update known replica list... "
+			    "disconnecting new replica(ip:%s port:%d "
+			    "guid:%lu replica_id: %s)\n", replica->ip, replica->port,
+			    replica->zvol_guid, replica->replica_id);
+			goto error_out_replica;
+		}
+	}
+
 	MTX_UNLOCK(&replica->r_mtx);
```

Comment:
why we need replica lock here?

Reply:
as discussed we are doing these things after creating data connection thread so it is required.